src/tachimawari/imu/node/imu_node.cpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2021 Ichiro ITS
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "tachimawari/imu/node/imu_node.hpp"
+
+#include "rclcpp/rclcpp.hpp"
+#include "kansei_interfaces/msg/unit.hpp"
+#include "keisan/keisan.hpp"
+#include "tachimawari/imu/node/imu_provider.hpp"
+
+namespace tachimawari
+{
+
+namespace imu
+{
+
+ImuNode::ImuNode(rclcpp::Node::SharedPtr node, std::shared_ptr<ImuProvider> imu_provider)
+: imu_provider(imu_provider)
+{
+  unit_publisher = node->create_publisher<kansei_interfaces::msg::Unit>(
+    get_node_prefix() + "/unit", 10);
+}
+
+void ImuNode::update_imu()
+{
+  auto unit_msg = kansei_interfaces::msg::Unit();
+
+  keisan::Vector<3> gyro = imu_provider->get_gyro();
+  keisan::Vector<3> accelero = imu_provider->get_accelero();
+
+  unit_msg.gyro = {static_cast<float>(gyro[0]), static_cast<float>(gyro[1]),
+    static_cast<float>(gyro[2])};
+  unit_msg.accelero = {static_cast<float>(accelero[0]), static_cast<float>(accelero[1]),
+    static_cast<float>(accelero[2])};
```

Comment:
you can solve this issue (https://github.com/ichiro-its/keisan/issues/56) to prevent casting it to floats. btw, can't we just pass the value dirrectly? without static casted it first?

Reply:
Maybe this will explain the whole `static_cast` usage. I use it just to make sure that the source variables haven't the same type as target variables, so those need to be cast.
However, I will consider this again and revise this because you pay attention about this. Possibly I will change the entire `static_cast` usage whenever there is no big problem when those variables aren't cast.

Reply:
done